recipes/views.py

```diff
@@ -167,7 +167,12 @@ def perform_create(self, serializer):
         serializer.save(author=self.request.user)
 
 
-class RetrieveUpdateDestroyRecipeView(RetrieveUpdateDestroyAPIView):
+class RecipeDetailView(RetrieveUpdateDestroyAPIView):
     serializer_class = RecipeSerializer
     queryset = Recipe.objects.all()
-    permission_classes = (permissions.IsAuthenticated, IsOwnerOrIsAdmin)
+    permission_classes = (IsOwnerOrIsAdmin, )
+
+    def get_permissions(self):
+        if self.request.method == 'GET':
```

Comment:
Let's allow`'GET', 'HEAD', 'OPTIONS'` methods. You can use `rest_framework.permissions.SAFE_METHODS`